modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/TableColumnHeaderTest.java

```diff
@@ -0,0 +1,146 @@
+                        return new TableHeaderRow(this) {
+                            @Override
+                            protected NestedTableColumnHeader createRootHeader() {
+                                return new NestedTableColumnHeader(null) {
+                                    @Override
+                                    protected TableColumnHeader createTableColumnHeader(TableColumnBase col) {
+                                        tableColumnHeader = new MyTableColumnHeader(column);
+                                        return col == null || col.getColumns().isEmpty() || col == getTableColumn() ?
+                                                tableColumnHeader :
+                                                new NestedTableColumnHeader(col);
+
+                                    }
+                                };
+                            }
+                        };
+                    }
+                };
+            }
+        };
+
+        tableView.getColumns().add(column);
+
+        Toolkit tk = Toolkit.getToolkit();
+
+        Scene scene = new Scene(tableView);
+        Stage stage = new Stage();
+        stage.setScene(scene);
+        stage.setWidth(500);
+        stage.setHeight(400);
+        stage.centerOnScreen();
+        stage.show();
+
+        tk.firePulse();
+
+        double width = column.getWidth();
+        tableColumnHeader.resizeCol();
+        assertEquals("Width must be the same",
+                width, column.getWidth(), 0.001);
+
+        EventType<TableColumn.CellEditEvent<Person, String>> eventType = TableColumn.editCommitEvent();
+        column.getOnEditCommit().handle(new TableColumn.CellEditEvent<Person, String>(
+                tableView, new TablePosition<Person, String>(tableView, 0, column), (EventType) eventType, "This is a big text inside that column"));
+        tableColumnHeader.resizeCol();
+        assertTrue("Column width must be greater",
+                width < column.getWidth());
+
+        column.getOnEditCommit().handle(new TableColumn.CellEditEvent<Person, String>(
+                tableView, new TablePosition<Person, String>(tableView, 0, column), (EventType) eventType, "small"));
+        column.getOnEditCommit().handle(new TableColumn.CellEditEvent<Person, String>(
+                tableView, new TablePosition<Person, String>(tableView, 1, column), (EventType) eventType, "small"));
+        column.getOnEditCommit().handle(new TableColumn.CellEditEvent<Person, String>(
+                tableView, new TablePosition<Person, String>(tableView, 2, column), (EventType) eventType, "small"));
+        column.getOnEditCommit().handle(new TableColumn.CellEditEvent<Person, String>(
+                tableView, new TablePosition<Person, String>(tableView, 3, column), (EventType) eventType, "small"));
+
+        tableColumnHeader.resizeCol();
+        assertTrue("Column width must be smaller",
+                width > column.getWidth());
```

Comment:
same as above but the other way round.

Until, the tests verified that the column width is effected by cell content and goes into the course direction of what is expected. What's still missing are tests that cover the complete specification (we now have one :), that is verify that/how resize is indeed depend on

- header content 
- maxRows 

Plus maybe that custom implementations are respected: f.i. doing nothing, making them constant, header only